packages/client/src/game/Game.ts

```diff
@@ -0,0 +1,362 @@
+
+        const cb = () => {
+          clickUno = true;
+          console.log('У юзера осталась одна карта');
+        };
+        /* Подписка на событие клика по кнопке UNO */
+        this.on('click uno', cb);
+        /* Если за 1,5 сек клика не было, игроку выдаются 2 карты на руку и ход переходит к след. игроку */
+        await sleep(1500, () => {
+          if (!clickUno) {
+            console.log('Юзер не нажал на кнопку');
+            this.takeCard(2);
+            return;
+          } else {
+            this.off('click uno', cb);
+          }
+        });
+      }
+    }
+
+    /* Проверяем, была ли выложена спец.карта. Если да, выполняем действие карты */
+    this.checkCard(movedCard);
+
+    this.moveLine();
+  }
+
+  checkCard(card: CardType) {
+    if (card.action === 'draw two') {
+      /* Следующий игрок берёт две карты и пропускает ход */
+      this.skipMove();
+      this.takeCard(2);
+    }
+
+    if (card.action === 'wild') {
+      /* Игрок выбирает новый цвет */
+      this.activeColor =
+        cardColors[Math.floor(Math.random() * cardColors.length)];
+      this.table.setColor(this.activeColor);
+    }
+
+    if (card.action === 'wild draw four') {
+      /* Игрок выбирает новый цвет, а следующий игрок берёт 4 карты */
+      this.activeColor =
+        cardColors[Math.floor(Math.random() * cardColors.length)];
+      this.table.setColor(this.activeColor);
+      this.skipMove();
+      this.takeCard(4);
+    }
+
+    if (card.action === 'reverse') {
+      /* Очёрёдность хода меняется в обратную сторону */
+      /* Но если два игрока, то второй игрок пропускает ход */
+      if (this.players.length === 2) {
+        this.skipMove();
+      } else {
+        this.clockwiseMovement = this.clockwiseMovement === true ? false : true;
+      }
+    }
+
+    if (card.action === 'skip') {
+      /* Следующий игрок пропускает ход */
+      this.skipMove();
+    }
+  }
+
+  takeCard(countCards: number) {
+    /* Обновление закрытой колоды когда карты заканчиваются */
+    this.table.renewClosePack();
+
+    const cards = this.table.giveCards(countCards);
+    const activePlayer = this.getActivePlayer();
+
+    activePlayer.addCards(cards);
+
+    /* Если берётся 1 карта и её можно выложить на стол, то игрок должен это сделать */
+    if (countCards === 1) {
+      if (compareCards(cards[0], this.table.getUpcard(), this.activeColor)) {
+        /* Если ход бота, делаем за него ход */
+        if (this.players[this.activePlayerId].isBot) {
+          console.log('Бот через секунду скинет карту');
+          const card = activePlayer.cards.at(-1);
+
+          if (card) {
+            setTimeout(() => {
+              this.discardCard(card);
+            }, 1000);
+          }
+        } else {
+          /* Если ход юзера, ждём клик */
+          console.log('Можно скинуть карту');
+        }
+      } else {
+        /* Если карта не подходит, переход хода */
+        this.moveLine();
+      }
+    }

```

Comment:
Ещё раз про уровни вложенности. Например давай тут я помогу
```suggestion
    if (countCards !== 1) {
      return;
    }

    // Тут надо дават правильное название compareCards-у
    const isCardsSomething = compareCards(cards[0], this.table.getUpcard(), this.activeColor)
    if (!isCardsSomething) {
      // Если карта не подходит, переход хода
      this.moveLine();
      return;
    }

    const { isBot } = this.players[this.activePlayerId];
    if (!isBot) {
      // Если ход юзера, ждём клик
      console.log('Можно скинуть карту');
      return;
    }

    // Если ход бота, делаем за него ход
    console.log('Бот через секунду скинет карту');

    const card = activePlayer.cards.at(-1);
    if (!card) {
      return;
    }

    setTimeout(() => {
      this.discardCard(card);
    }, 1000);
```

Reply:
Т.е. надо стараться писать код как можно плавнее, это сильно упрощает когнитивную сложность кода.